src/Compilers/CSharp/Portable/Symbols/Synthesized/GeneratedNames.cs

```diff
@@ -465,6 +465,15 @@ internal static string MakeSynthesizedInlineArrayName(int arrayLength, int gener
             return (generation > 0) ? name + GenerationSeparator + generation : name;
         }
 
+        internal static string MakeSynthesizedReadOnlyListName(int generation)
+        {
+            Debug.Assert((char)GeneratedNameKind.ReadOnlyListType == 'z');
+            var name = "<>z__ReadOnlyList";
+
+            // Synthesized list types need to have unique name across generations because they are not reused.
```

Comment:
what's a generation?  something about EnC?

Reply:
Yes, a generation is an iteration of EnC.